homeassistant/components/lg_soundbar/media_player.py

```diff
@@ -1,27 +1,42 @@
 """Support for LG soundbars."""
 from __future__ import annotations
 
+import logging
+
 import temescal
 
 from homeassistant.components.media_player import (
     MediaPlayerEntity,
     MediaPlayerEntityFeature,
 )
-from homeassistant.const import STATE_ON
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_UNIQUE_ID, STATE_ON
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the LG platform."""
-    if discovery_info is not None:
-        add_entities([LGDevice(discovery_info)])
+    """Set up media_player from a config entry created in the integrations UI."""
+    config = config_entry.data
+    # Update our config.
+    if config_entry.options:
+        hass.data[DOMAIN][config_entry.entry_id].update(config_entry.options)
+    if config[CONF_HOST] is not None and config[CONF_PORT] is not None:
+        info = {
+            CONF_UNIQUE_ID: config_entry.unique_id,
+            CONF_HOST: config[CONF_HOST],
+            CONF_PORT: config[CONF_PORT],
+        }
+        entities = [LGDevice(info)]
+        async_add_entities(entities)
```

Comment:
`hass.data[DOMAIN]` is never initialized.

```suggestion
    info = {
        CONF_UNIQUE_ID: config_entry.unique_id,
        CONF_HOST: config[CONF_HOST],
        CONF_PORT: config[CONF_PORT],
    }
    entities = [LGDevice(info)]
    async_add_entities(entities)
```